#!/usr/bin/env python3

PACKAGE = "sphero_stage_2"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# gen.add("int_param",    int_t,    0, "An Integer parameter", 50,  0, 100)

# awais weights param
# gen.add("align_vel_weights", double_t, 0, "A double parameter",       0.8, 0.0,   2.0)
# gen.add("cohesion_vel_weights", double_t, 0, "A double parameter",    0.09, 0.0,   2.0)
# gen.add("separation_vel_weights", double_t, 0, "A double parameter",  1.5, 0.0,   3.0)

# my weights param
gen.add("nav_weight", double_t, 0, "A double parameter",    0.15, 0.0,   6.0)
gen.add("rep_weight", double_t, 0, "A double parameter",    0.8, 0.0,   5.0) 


# awais other params
# gen.add("neighbours_dist", double_t, 0, "A double parameter",    1.0, 0.0,   2.0)
# gen.add("neighbours_angle", double_t, 0, "A double parameter",    3.147, 0.0,   5.0)
# gen.add("separation_dist", double_t, 0, "A double parameter",    0.4, 0.0,   2.0)
# gen.add("max_force", double_t, 0, "A double parameter",    10.0, 0.0,   20.0)

# my other params
gen.add("max_speed", double_t, 0, "A double parameter",    0.6, 0.0,   2.0)
gen.add("slowing_radius", double_t, 0, "A double parameter",    1.0, 0.0,   2.0)
# gen.add("search_radius", double_t, 0, "A double parameter",    2.0, 0.0,   4.0)


# gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)


exit(gen.generate(PACKAGE, "sphero_stage_2", "obstacle_avoidance_2"))
